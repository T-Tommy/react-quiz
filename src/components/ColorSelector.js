import React from 'react';
import PropTypes from 'prop-types';

function ColorSelector({ color, background, handleChange }) {
  return (
    <>
      <div>
        <label>
          Text Color:{ ' ' }
          <input
            type="color"
            name="color"
            value={ color }
            onChange={ handleChange }
          />
        </label>
      </div>
      <div>
        <label>
          Background Color:{ ' ' }
          <input
            type="color"
            name="background"
            value={ background }
            onChange={ handleChange }
          />
        </label>
      </div>
    </>
  );
}

ColorSelector.propTypes = {
  color: PropTypes.string.isRequired,
  background: PropTypes.string.isRequired,
  handleChange: PropTypes.func.isRequired
};

export default ColorSelector;
