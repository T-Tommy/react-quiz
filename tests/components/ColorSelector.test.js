import React from 'react';
import { shallow } from 'enzyme';
import ColorSelector from '../../src/components/ColorSelector';

describe('ColorSelector component', () => {
  it('renders the ColorSelector', () => {
    const wrapper = shallow(<ColorSelector color="#000000" background="#FFFFFF" handleChange={ jest.fn } />);
    expect(wrapper).toMatchSnapshot();
  });
});
