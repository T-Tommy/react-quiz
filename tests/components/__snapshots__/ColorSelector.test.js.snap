// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorSelector component renders the ColorSelector 1`] = `
<Fragment>
  <div>
    <label>
      Text Color:
       
      <input
        name="color"
        onChange={[Function]}
        type="color"
        value="#000000"
      />
    </label>
  </div>
  <div>
    <label>
      Background Color:
       
      <input
        name="background"
        onChange={[Function]}
        type="color"
        value="#FFFFFF"
      />
    </label>
  </div>
</Fragment>
`;
